### PR TITLE
add requires cert-manager: v1

### PR DESCRIPTION
**Pull request by s12chung**
**What type of PR is this?**
Chore

**What this PR does/ why we need it**:
Ensure that cert-manager upgrade works, by ensuring that adding wait for cert-manager v1

**Which issue(s) this PR fixes**:
D2IQ-72804

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Checklist**

* [ ] The commit message explains the changes and why are needed.
* [ ] The code builds and passes lint/style checks locally.
* [ ] The relevant subset of integration tests pass locally.
* [ ] The core changes are covered by tests.
* [ ] The documentation is updated where needed.

### CODE REVIEW

**Review by gracedo — APPROVED** [review]
hmm does dex need dependency on cert-manager? we had to make the cert-manager api updates in the chart but the addon doesn't have the dependency.. should it 🤔  (I'm cross-referencing this jira https://jira.d2iq.com/browse/D2IQ-70659 which included all the addons we needed to update with cert-manager v1)

**Comment by s12chung** [comment]
pinged @jr0d

**Comment by jr0d** [comment]
@s12chung lgtm, but I think if we want to be paranoid, any addon that requires an ingresscontroller, dex, dex-k8s-authenticater, traefik-forward-auth, kube-oidc-proxy, etc also rely on cert-manager. I think we can hold off on this for now, but keep it staged in case we run into the dependency again.

File: addons/dex/dex.yaml
Comment (gracedo) [inline addons/dex/dex.yaml:29]: dupe
```suggestion
```

Comment (s12chung) [reply addons/dex/dex.yaml:29]: I believe we need this to match different addons:

https://github.com/mesosphere/kubernetes-base-addons/blob/f7d5de0f1be7249cc8f1f02c94047f14ae5ecf44/addons/gcpdiskprovisioner/gcpdiskprovisioner.yaml#L14-L18

Comment (joejulian) [reply addons/dex/dex.yaml:29]: No, not a dup
There are two separate dependencies being matched.

**Review by joejulian — APPROVED** [review]

**Pull request merged**
